### PR TITLE
wr-194: article improvement suggestions

### PR DESCRIPTION
**Pull request by artemmatiushenko1**
This pull request adds an article improvement suggestion feature.

- Added API endpoint to generate suggestions using openAI (returns title, description, and priority of suggestion)
- Added corresponding component to frontend

image
image
image

### CODE REVIEW

File: backend/src/packages/articles/article.service.ts [hunk 1]
```diff
@@ -73,10 +76,13 @@ class ArticleService implements IService {
       return null;
     }
 
-    const [firstParsedGenre] =
-      safeJSONParse<DetectedArticleGenre[]>(genresJSON) ?? [];
+    const parsedGenres = safeJSONParse<DetectedArticleGenre[]>(genresJSON);
 
-    return firstParsedGenre ?? null;
+    if (Array.isArray(parsedGenres) && parsedGenres[0]) {
+      return parsedGenres[0];
```
Comment (andanf-e) [inline backend/src/packages/articles/article.service.ts]: **B31**
```suggestion
      return parsedGenres[FIRST_ITEM_INDEX];
```

File: backend/src/packages/articles/article.service.ts [hunk 2]
```diff
@@ -172,6 +185,51 @@ class ArticleService implements IService {
...
+      safeJSONParse<ArticleImprovementSuggestion[]>(suggestionsJSON);
+
+    if (Array.isArray(parsedSuggestions)) {
+      return parsedSuggestions;
+    }
+
+    return null;
+  }
+
+  public async getArticleImprovementSuggesstions(
```
Comment (andanf-e) [inline backend/src/packages/articles/article.service.ts]:
```suggestion
  public async getArticleImprovementSuggestions(
```

File: frontend/src/pages/article/components/article-improvement-suggestions/article-improvement-suggestions.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  useEffect(() => {
+    void dispatch(articlesActions.getImprovementSuggestionsBySession(id));
+  }, [dispatch, id]);
+
+  const handleSuggestImprovements = (): void => {
+    void dispatch(articlesActions.getImprovementSuggestions(id));
+  };
+
+  const showSuggestImprovementButton =
+    (!improvementSuggestions || improvementSuggestions.length <= 0) &&
```
Comment (andanf-e) [inline frontend/src/pages/article/components/article-improvement-suggestions/article-improvement-suggestions.tsx]:
```suggestion
    !improvementSuggestions?.length &&
```

File: frontend/src/pages/article/components/article-improvement-suggestions/libs/enums/article-suggestion-priority.enum.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const ArticleSuggestionPriority = {
+  Low: 1,
+  Medium: 2,
+  High: 3,
```
Comment (andanf-e) [inline frontend/src/pages/article/components/article-improvement-suggestions/libs/enums/article-suggestion-priority.enum.ts]: **B23**
```suggestion
  LOW: 1,
  MEDIUM: 2,
  HIGH: 3,
```

File: shared/src/packages/articles/libs/enums/article-improvement-suggestion-priority.enum.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const ArticleImprovementSuggestionPriority = {
+  Low: 1,
+  Medium: 2,
+  High: 3,
```
Comment (andanf-e) [inline shared/src/packages/articles/libs/enums/article-improvement-suggestion-priority.enum.ts]: Same about **B23**,
And if you declare it in shared, then I am not sure you need to duplicate it on frontend, maybe just reuse this enum?
```suggestion
  Low: 1,
  Medium: 2,
  High: 3,
```

Comment (artemmatiushenko1) [reply shared/src/packages/articles/libs/enums/article-improvement-suggestion-priority.enum.ts]: Just forgot to delete it 😓

**Review by andanf-e — DISMISSED** [review]

**Review by andanf-e — APPROVED** [review]

**Pull request merged**